### PR TITLE
Updates Voteable things to implement Reporting

### PR DESCRIPTION
**Pull request by corylulu**
Allows for voteable things to be reported.

### CODE REVIEW

File: RedditSharp/Things/VotableThing.cs [hunk 1]
```diff
@@ -158,5 +169,46 @@ public void ClearVote()
...
+            var request = WebAgent.CreatePost(ReportUrl);
+            var stream = request.GetRequestStream();
+
+            string reportReason;
+            switch (reportType)
+            {
+                case ReportType.Spam: 
+                    reportReason = "spam"; break;
+                case ReportType.VoteMinipulation: 
+                    reportReason = "vote minipulation"; break;
```
Comment (SirCmpwn) [inline RedditSharp/Things/VotableThing.cs]: That's not how "manipulation" is spelled.

File: RedditSharp/Things/VotableThing.cs [hunk 2]
```diff
@@ -158,5 +169,46 @@ public void ClearVote()
             var response = request.GetResponse();
             var data = WebAgent.GetResponseString(response.GetResponseStream());
         }
+
+        public void Report(ReportType reportType)
+        {
+            Report(reportType, "");
+        }
+
+        public void Report(ReportType reportType, string otherReason)
```
Comment (SirCmpwn) [inline RedditSharp/Things/VotableThing.cs]: `otherReason` should default to `null`.

Comment (corylulu) [inline RedditSharp/Things/VotableThing.cs]: Wasn't sure if you wanted it to be compatible with earlier .NET's that don't have default parameters.

Comment (SirCmpwn) [inline RedditSharp/Things/VotableThing.cs]: There are already default parameters all over the place in RedditSharp.

Comment (corylulu) [inline RedditSharp/Things/VotableThing.cs]: Okay, wasn't paying much attention, just added it really quick and didn't see any default parameters in my open documents, so I changed it to be an overload. I resubmitted the fixes though.

**Pull request closed without merging**